modules/apps/27-interchain-accounts/module.go

```diff
@@ -144,13 +144,18 @@ func (am AppModule) RegisterServices(cfg module.Configurator) {
 		hosttypes.RegisterQueryServer(cfg.QueryServer(), am.hostKeeper)
 	}
 
-	m := controllerkeeper.NewMigrator(am.controllerKeeper)
-	if err := cfg.RegisterMigration(types.ModuleName, 1, m.AssertChannelCapabilityMigrations); err != nil {
+	controllerMigrator := controllerkeeper.NewMigrator(am.controllerKeeper)
+	if err := cfg.RegisterMigration(types.ModuleName, 1, controllerMigrator.AssertChannelCapabilityMigrations); err != nil {
 		panic(fmt.Sprintf("failed to migrate interchainaccounts app from version 1 to 2: %v", err))
 	}
 
-	hostm := hostkeeper.NewMigrator(am.hostKeeper)
-	if err := cfg.RegisterMigration(types.ModuleName, 2, hostm.MigrateParams); err != nil {
+	hostMigrator := hostkeeper.NewMigrator(am.hostKeeper)
+	if err := cfg.RegisterMigration(types.ModuleName, 2, func(ctx sdk.Context) error {
+		if err := hostMigrator.MigrateParams(ctx); err != nil {
+			return err
+		}
+		return controllerMigrator.MigrateParams(ctx)
```

Comment:
Isn't that checked above? In lines 136-145

Reply:
I think @colin-axner means that `hostMigrator.MigrateParams` and `controllerMigrator.MigrateParams` both use the host and controller keeper respectively and, by the look of the code, both `hostkeeper.NewMigrator` and `controllerkeeper.NewMigrator` could potentially be passed a `nil` keeper...

Reply:
Yeah, the code here code use a little restructuring imo for better readability/maintainability. 

The existing code was fine because the keeper is nil checked in the controller migration [here](https://github.com/cosmos/ibc-go/blob/main/modules/apps/27-interchain-accounts/controller/keeper/migrations.go#L28) for `AssertChannelCapabilityMigrations`. 

It may be nicer to move the instantiation of the migrator types into the nil checks above at the top of this function.

Reply:
Is the worry here that a keeper won't be initialized? Or incorrect wiring will take place? I mean, there's gotta be some sort of error if keeper is nil right? Sorry, I'm lost here. Or do we want keeper to be able to be nil?

Reply:
Both the controller and the host keeper are allowed to be `nil` because chains might choose to to use one or the other submodule. See [this section of the documentation](https://ibc.cosmos.network/main/apps/interchain-accounts/integration.html#using-submodules-exclusively) for more info.

Reply:
I've moved the check inside the migration functions